### PR TITLE
feat: LEAP-47: Move datamanager into the monorepo

### PR DESCRIPTION
**Pull request by juliosgarbi**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Frontend

### Describe the reason for change
The basis of this solution surrounds the collapsing of our multi-repo frontend code to a mono-repos in LabelStudio.

Orchestration and tooling will be handled by [NX.dev](http://nx.dev/), which allows us to maintain our code as a mono-repo architecture in a much more focused manner and with less boilerplate, configuration juggling and maintenance burden.

#### What is the new behavior?
It's getting datamanager to monorepo structure
Allowing main app to consume this lib, and update any import references to support this.

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
+export default {
+  displayName: 'datamanager',
+  preset: '../../jest.preset.js',
```
Comment (yyassi-heartex) [inline jest.config.ts:4]: path goes out of scope of the project, is that intentional?

File: .eslintrc.js [hunk 2]
```diff
@@ -16,7 +16,9 @@ module.exports = {
   extends: [
     "eslint:recommended",
     "plugin:react/recommended",
-    "plugin:@typescript-eslint/recommended"
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@nx/react",
+    "../../.eslintrc.json"
```
[the remark this answers is not in the document]
Comment (juliosgarbi) [reply .eslintrc.js:21]: Yeah, it's intentional, basically each lib or app has their only eslint config, also all of them share the same config with the monorepo config.

File: jest.config.ts (hunk 1 above)
Comment (juliosgarbi) [reply jest.config.ts:4]: Yeah, in general the idea behind of NX is that libs and apps share the same scope, and all these configs are on NX root folder

File: tsconfig.spec.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "extends": "../../../../../Desktop/untitled folder/tsconfig.json",
```
[the remark this answers is not in the document]
Comment (juliosgarbi) [reply tsconfig.spec.json:2]: That is totally wrong, my bad!

**Review by yyassi-heartex — APPROVED** [review]

**Review by bmartel — APPROVED** [review]

**Pull request merged**